File: OtakuNest.Client/src/app/core/params/query-string-parameters.ts
export abstract class QueryStringParameters {
    private readonly maxPageSize = 50;

    pageNumber: number = 1;

    private _pageSize: number = 10;
    get pageSize(): number {
        return this._pageSize;
    }
    set pageSize(value: number) {
        this._pageSize = value > this.maxPageSize ? this.maxPageSize : value;
    }

    orderBy?: string = 'Id';
}

File: OtakuNest.Common/Parameters/QueryStringParameters.cs
namespace OtakuNest.Common.Parameters
{
    public abstract class QueryStringParameters
    {
        private const int MaxPageSize = 50;
        private int _pageSize = 10;
        private int _pageNumber = 1;

        public int PageNumber
        {
            get => _pageNumber;
            set
            {
                if (value <= 0)
                    throw new ArgumentException("PageNumber must be greater than 0");
                _pageNumber = value;
            }
        }

        public int PageSize
        {
            get => _pageSize;
            set
            {
                if (value <= 0)
                    throw new ArgumentException("PageSize must be greater than 0");
                _pageSize = (value > MaxPageSize) ? MaxPageSize : value;
            }
        }

        public string? OrderBy { get; set; } = "Id";
    }
}
